types/osrm/index.d.ts

```diff
@@ -0,0 +1,587 @@
+ */
+interface TripOptions extends Options {
+    /**
+     * Return route is a roundtrip. (optional, default true)
+     */
+    roundtrip?: boolean;
+    /**
+     * Return route starts at any coordinate. Can also be first. (optional, default any)
+     */
+    source?: string;
+    /**
+     * Return route ends at any coordinate. Can also be last. (optional, default any)
+     */
+    destination?: string;
+    /**
+     * Return route steps for each route. (optional, default false)
+     */
+    steps?: boolean;
+    /**
+     * Return annotations for each route leg for duration, nodes, distance, weight, datasources and/or speed.
+     * Annotations can be false or true (no/full annotations) or an array of strings with duration, nodes, distance, weight, datasources, speed. (optional, default false)
+     */
+    annotations?: boolean | string[];
+    /**
+     * Returned route geometry format (influences overview and per step). Can also be geojson. (optional, default polyline)
+     */
+    geometries?: string;
+    /**
+     * Add overview geometry either full, simplified (optional, default simplified)
+     */
+    overview?: string;
+}
+
+interface RouteResults {
+    waypoints: Waypoint[];
+    routes: Route[];
+}
+
+interface NearestResults {
+    waypoints: Waypoint[];
+}
+
+interface TableResults {
+    durations: Duration[][];
+    sources: Waypoint[];
+    destinations: Waypoint[];
+}
+
+interface MatchResults {
+    tracepoints: MatchWaypoint[];
+    matchings: MatchRoute[];
+}
+
+interface TripResults {
+    waypoints: TripWaypoint[];
+    trips: Route[];
+}
+
+/**
+ * The OSRM method is the main constructor for creating an OSRM instance.
+ * An OSRM instance requires a .osrm network, which is prepared by the OSRM Backend C++ library.
+ *
+ * https://github.com/Project-OSRM/node-osrm/blob/master/docs/api.md
+ */
+declare class OSRM {
+    constructor(name: string);
+    /**
+     * shortest path between given coordinates
+     */
+    route(options: RouteOptions, callback: (err: Error, results: RouteResults) => void): void;
+    /**
+     * returns the nearest street segment for a given coordinate
+     */
+    nearest(options: NearestOptions, callback: (err: Error, results: NearestResults) => void): void;
+
+    /**
+     * computes distance tables for given coordinates
+     */
+    table(options: TableOptions, callback: (err: Error, results: TableResults) => void): void;
+    /**
+     * matches given coordinates to the road network
+     */
+    match(options: MatchOptions, callback: (err: Error, results: MatchResults) => void): void;
+    /**
+     * Compute the shortest trip between given coordinates
+     */
+    trip(options: TripOptions, callback: (err: Error, results: TripResults) => void): void;
+    /**
+     * Return vector tiles containing debugging info
+     */
+    tile(options: TileOptions | Tile, callback: (err: Error, results: Buffer) => void): void;
+}
+
+declare namespace OSRM {
```

Comment:
If you would like consumers of this declaration file to be able to use your types and interfaces, you should include them within this namespace and export them.

Reply:
Agreed, I'll include the interfaces in the namespace 👍

Reply:
@uniqueiniquity Modified the Typescript definition to reflect your comment.